go/stats/main.go

```diff
@@ -167,43 +171,46 @@ type eduroamtype struct{}
 func (s eduroamtype) Test(source interface{}, ctx context.Context) {
 
 	t := StatsdClient.NewTiming()
-	packet := radius.New(radius.CodeAccessRequest, []byte(source.(pfconfigdriver.AuthenticationSourceEduroam).RadiusSecret))
-	UserName_SetString(packet, "tim")
-	UserPassword_SetString(packet, "12345")
-	client := radius.DefaultClient
-	ctx2, cancel := context.WithTimeout(ctx, 5*time.Second)
-	defer cancel()
-	response, err := client.Exchange(ctx2, packet, source.(pfconfigdriver.AuthenticationSourceEduroam).Server1Address+":1812")
+	if source.(pfconfigdriver.AuthenticationSourceEduroam).TestUser != "" {
+		packet := radius.New(radius.CodeAccessRequest, []byte(source.(pfconfigdriver.AuthenticationSourceEduroam).RadiusSecret))
+		UserName_SetString(packet, source.(pfconfigdriver.AuthenticationSourceEduroam).TestUser)
+		UserPassword_SetString(packet, source.(pfconfigdriver.AuthenticationSourceEduroam).TestPassword)
+		client := radius.DefaultClient
+		ctx2, cancel := context.WithTimeout(ctx, 5*time.Second)
+		defer cancel()
+		response, err := client.Exchange(ctx2, packet, source.(pfconfigdriver.AuthenticationSourceEduroam).Server1Address+":1812")
 
-	if err != nil {
-		StatsdClient.Gauge("source."+source.(pfconfigdriver.AuthenticationSourceEduroam).Type+"."+source.(pfconfigdriver.AuthenticationSourceEduroam).PfconfigHashNS+"1", 0)
-	} else {
-		StatsdClient.Count("source."+source.(pfconfigdriver.AuthenticationSourceEduroam).Type+"."+source.(pfconfigdriver.AuthenticationSourceEduroam).PfconfigHashNS+"1", 1)
-		if response.Code == radius.CodeAccessAccept {
-			fmt.Println("Accepted")
+		if err != nil {
+			StatsdClient.Gauge("source."+source.(pfconfigdriver.AuthenticationSourceEduroam).Type+"."+source.(pfconfigdriver.AuthenticationSourceEduroam).PfconfigHashNS+"1", 0)
 		} else {
-			fmt.Println("Denied")
+			StatsdClient.Count("source."+source.(pfconfigdriver.AuthenticationSourceEduroam).Type+"."+source.(pfconfigdriver.AuthenticationSourceEduroam).PfconfigHashNS+"1", 1)
+			if response.Code == radius.CodeAccessAccept {
```

Comment:
Would be useful to record whether it was accepted or not for more in-depth troubleshooting if needed